File: contacting-conf-web/src/main/java/nl/ing/api/contacting/conf/domain/model/permission/ContactingConfigVO.java
package nl.ing.api.contacting.conf.domain.model.permission;

import java.util.Arrays;
import java.util.Set;
import java.util.stream.Collectors;

public record ContactingConfigVO(String key, String values) {

    public Set<String> valuesAsSet() {
        return Arrays.stream(values.split(","))
                .collect(Collectors.toSet());
    }
}

File: contacting-conf-web/src/main/scala/nl/ing/api/contacting/conf/domain/ContactingConfigVO.scala
package nl.ing.api.contacting.conf.domain

import scala.util.Try

case class ContactingConfigVO(key: String, values: String) {
  /**
    * will split the values by ",", so it will give a set of values
    *
    * @return
    */
  def valuesAsSet(): Set[String] = {
    values.split(",").toSet
  }

  private def toBoolean(default: Boolean): Boolean=
    Try(values.toBoolean).getOrElse(default)

  def booleanValueWithFalseDefault: Boolean =
    toBoolean(false)

  def booleanValueWithTrueDefault: Boolean =
    toBoolean(true)

}
